### PR TITLE
refactor(editor): Migrate `EventDestinationSettingsModal.ee.vue` to composition API (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue [hunk 1]
```diff
@@ -22,338 +20,336 @@ import {
...
-			testMessageResult: false,
-			LOG_STREAM_MODAL_KEY,
-		};
+import {
+	webhookModalDescription,
+	sentryModalDescription,
+	syslogModalDescription,
+} from './descriptions.ee';
+
+defineOptions({ name: 'EventDestinationSettingsModal' });
```
Comment (RicardoE105) [inline packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue:51]: nitpick: I do not think we need this.

File: packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue [hunk 2]
```diff
@@ -22,338 +20,336 @@ import {
...
+
+function onLabelChange(value: string) {
+	onInput();
+	headerLabel.value = value;
+	nodeParameters.value.label = value;
+}
+
+function setupNode(options: MessageEventBusDestinationOptions) {
+	workflowsStore.removeNode(node.value);
+	ndvStore.activeNodeName = options.id ?? 'thisshouldnothappen';
```
Comment (RicardoE105) [inline packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue:182]: No idea how this made into the code "thisshouldnothappen" 😆

File: packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue [hunk 3]
```diff
@@ -22,338 +20,336 @@ import {
...
+					syslogDestination.app_name !== ''
+				);
 			}
-		},
-	},
-});
+			return false;
+		};
+
+		useTelemetry().track('User updated log streaming destination', {
```
Comment (RicardoE105) [inline packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue:339]: nitpick: move composable up where the other composables are defined

File: packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue [hunk 4]
```diff
@@ -22,338 +20,336 @@ import {
...
+			is_active: destination.enabled,
+		});
+	}
+}
+
+function callEventBus(event: string, data: unknown) {
+	if (eventBus) {
+		eventBus.emit(event, data);
+	}
+}
```
Comment (RicardoE105) [inline packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue:352]: Replace all $locale in the template with i18n

File: packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue (hunk 2 above)
Comment (netroy) [reply packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue:182]: doesn't seem to be from this PR though. https://github.com/n8n-io/n8n/pull/4860/files#diff-4c588129b99621a76498f97fa57f76a20ebda61480ceac0d4ac36a1682090d76R361

File: packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue (hunk 4 above)
Comment (netroy) [reply packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue:353]: I thought I did this already. Thanks for catching this 🙏🏽

File: packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue (hunk 1 above)
Comment (netroy) [reply packages/editor-ui/src/components/SettingsLogStreaming/EventDestinationSettingsModal.ee.vue:51]: as discussed on Slack, we'll remove all instances of `defineOptions` after we are done migrating all the components.

**Review by RicardoE105 — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.68.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.68.0)

**Pull request merged**